sentry/src/main/java/io/sentry/instrumentation/file/FileIOSpanManager.java

```diff
@@ -0,0 +1,110 @@
+package io.sentry.instrumentation.file;
+
+import io.sentry.IHub;
+import io.sentry.ISpan;
+import io.sentry.SpanStatus;
+import io.sentry.util.Pair;
+import io.sentry.util.Platform;
+import io.sentry.util.StringUtils;
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+final class FileIOSpanManager {
+
+  private final @Nullable ISpan currentSpan;
+  private final @Nullable File file;
+  private final @NotNull IHub hub;
+
+  private @Nullable Throwable throwable = null;
+  private @NotNull SpanStatus spanStatus = SpanStatus.OK;
+  private long byteCount;
+
+  static @Nullable ISpan startSpan(final @NotNull IHub hub, final @NotNull String op) {
+    final ISpan parent = hub.getSpan();
+    return parent != null ? parent.startChild(op) : null;
+  }
+
+  FileIOSpanManager(
+      final @Nullable ISpan currentSpan, final @Nullable File file, final @NotNull IHub hub) {
+    this.currentSpan = currentSpan;
+    this.file = file;
+    this.hub = hub;
+  }
```

Comment:
```suggestion
  FileIOSpanManager(
      final @Nullable ISpan currentSpan, final @Nullable File file, final @NotNull IHub hub) {
    this.currentSpan = currentSpan;
    this.file = file;
    this.hub = hub;
  }

  FileIOSpanManager(
          final @Nullable ISpan currentSpan, final @Nullable File file) {
    this(currentSpan, file, HubAdapter.getInstance());
  }
```

the ctor that accepts the hub is only used for testing

Reply:
unless this class is never gonna be used by clients, so then it makes sense to always use the very same hub

Reply:
See https://github.com/getsentry/sentry-java/pull/1826#discussion_r766445062